compiler/src/codegen/compcore.re

```diff
@@ -3478,6 +3482,13 @@ let compile_wasm_module =
       Option.is_none(Grain_utils.Config.memory_base^),
     );
 
+  // Our compiler generates code that should never trap, so enabling this allows
+  // Binaryen to perform more aggressive optimizations
+  let _ = Settings.set_traps_never_happen(true);
```

Comment:
Should we only enable this under our `Release` profile, just to help us catch any codegen errors?